src/renderer/components/+extensions/__tests__/extensions.test.tsx

```diff
@@ -103,39 +97,47 @@ describe("Extensions", () => {
 
     fireEvent.click(menuTrigger);
 
-    expect(res.getByText("Disable")).toHaveAttribute("aria-disabled", "false");
-    expect(res.getByText("Uninstall")).toHaveAttribute("aria-disabled", "false");
+    expect(await res.findByText("Disable")).toHaveAttribute("aria-disabled", "false");
+    expect(await res.findByText("Uninstall")).toHaveAttribute("aria-disabled", "false");
 
-    fireEvent.click(res.getByText("Uninstall"));
+    fireEvent.click(await res.findByText("Uninstall"));
 
     // Approve confirm dialog
-    fireEvent.click(res.getByText("Yes"));
+    fireEvent.click(await res.findByText("Yes"));
 
-    await waitFor(() => {
+    await waitFor(async () => {
       expect(extensionDiscovery.uninstallExtension).toHaveBeenCalled();
       fireEvent.click(menuTrigger);
-      expect(res.getByText("Disable")).toHaveAttribute("aria-disabled", "true");
-      expect(res.getByText("Uninstall")).toHaveAttribute("aria-disabled", "true");
+      expect(await res.findByText("Disable")).toHaveAttribute("aria-disabled", "true");
+      expect(await res.findByText("Uninstall")).toHaveAttribute("aria-disabled", "true");
     }, {
       timeout: 30000,
     });
   });
 
   it("disables install button while installing", async () => {
     const res = render(<Extensions />);
+    const url = "https://test.extensionurl/package.tgz";
 
     (fse.unlink as jest.MockedFunction<typeof fse.unlink>).mockReturnValue(Promise.resolve() as any);
 
     fireEvent.change(res.getByPlaceholderText("File path or URL", {
       exact: false,
     }), {
       target: {
-        value: "https://test.extensionurl/package.tgz",
+        value: url,
       },
     });
 
-    fireEvent.click(res.getByText("Install"));
-    expect(res.getByText("Install").closest("button")).toBeDisabled();
+    let resolveBuffer: (buf: Buffer) => void;
+
+    mockFetch.getOnce(url, new Promise<Buffer>(resolve => {
+      resolveBuffer = resolve;
+    }));
+
+    fireEvent.click(await res.findByText("Install"));
+    expect((await res.findByText("Install")).closest("button")).toBeDisabled();
```

Comment:
Brittle selector may cause test failure from irrelevant reason. Use eg. a data-attribute dedicated to only this purpose?

I acknowledge there's an aggravated discussion on this topic somewhere else.